components/ILIAS/Notifications/UserSettings/PushNotification.php

```diff
@@ -76,11 +76,12 @@ public function getInput(
         $pref = $user ? $this->retrieveValueFromUser($user) : [];
         $provider = [];
 
-        foreach ((new PushNotificationObjective(new NullConfig()))->getArtifacts() as $value) {
-            $provider[get_class($value)] = $field_factory->checkbox(
-                $value->getName($lng),
-                $value->getDescription($lng),
-            )->withValue(in_array(get_class($value), $pref));
+        foreach (require PushNotificationObjective::PATH() as $class) {
+            $obj = new $class();
```

Comment:
Can we always assume that all providers have exactly zero constructor parameters?

Reply:
I adapted that. Now we can